src/main/java/kth/iv1201/gohire/controller/PersonController.java

```diff
@@ -107,19 +108,30 @@ public List<ApplicantDTO> fetchApplicants(){
      * Changes the status of an application.
      * @param request DTO containing application change request data.
      * @return the changed and saved application.
+     * @throws LoggerException if there is a problem with logging an event.
+     * @throws ApplicationHandledException if the applicant has already been handled.
      */
     @PreAuthorize("hasRole('recruiter')")
     @PostMapping("/changeApplicationStatus")
     public ApplicantDTO changeApplicationStatus(@RequestBody @Valid ChangeApplicationStatusRequestDTO request)
-            throws LoggerException, ApplicationHandledException {
-        authenticateRequest(request.getUsername(), request.getPassword());
-        ApplicantDTO changedApplicant = personService.changeApplicantStatus(request);
-        Logger.logEvent("Recruiter " + request.getUsername() + " changed status of applicant " + changedApplicant.getFirstName() + " " +
-                changedApplicant.getLastName() + " to " + changedApplicant.getStatus() + ".");
-        return changedApplicant;
+            throws LoggerException, ApplicationHandledException, AuthenticationFailedException {
+        Authentication requestCredentialAuth;
+        try {
+            requestCredentialAuth = authenticateRequest(request.getUsername(), request.getPassword());
+        } catch (Exception e) {
+            throw new AuthenticationFailedException("Incorrect username or password.");
+        }
+        Authentication currentLoggedInRecruiterAuth = SecurityContextHolder.getContext().getAuthentication();
+        if (currentLoggedInRecruiterAuth.equals(requestCredentialAuth)) {
+            ApplicantDTO changedApplicant = personService.changeApplicantStatus(request);
+            Logger.logEvent("Recruiter " + request.getUsername() + " changed status of applicant " + changedApplicant.getFirstName() + " " +
+                    changedApplicant.getLastName() + " to " + changedApplicant.getStatus() + ".");
+            return changedApplicant;
+        } else {
+            throw new AuthenticationFailedException("Incorrect username or password.");
```

Comment:
Funderar på namnet på `AuthenticationFailedException`, det är alltså antingen om man skriver fel login-credentials eller om man skriver någon annans login-credentials än den man är inloggad som? Kanske kan heta typ `AuthenticationForLoggedInUserFailed` (eller nåt bättre som inte är superlångt 😅), och texten typ "Incorrect username or password, or logged in as wrong user" - så man fattar mer exakt vad det handlar om

Reply:
Fixat!